contents/docs/how-posthog-works/ingestion-pipeline.mdx

```diff
@@ -0,0 +1,172 @@
+
+For more detail on this step, check out [this file](https://github.com/PostHog/posthog/blob/master/plugin-server/src/worker/ingestion/event-pipeline/2-pluginsProcessEventStep.ts) from the `plugin-server` codebase.
+
+### 3. Person processing
+
+The next step in the ingestion pipeline is processing the Person who sent the event, which is determined by the `distinct_id` field.
+A number of different actions can happen here depending both on if we've seen this `distinct_id` before, as well as which type of event is being sent.
+
+This is one of the most complex steps in the entire pipeline, so to help make things easier we'll break down this step into a number of smaller sections:
+
+1. **Associate the event with a person**
+    1. `$identify` events
+    2. `$alias` events
+    3. All other events
+2. **Update person properties**
+
+#### 1. Associate the event with a person
+
+Based on which type of event is currently being processed, we perform a number of different steps.
+
+#### 1.1 - `$identify` events
+
+In the case of an `$identify` event, the first step is to use the `$distinct_id` and `$anon_distinct_id` fields that are sent on every event to determine what actions we will need to take.
+
+-   `$anon_distinct_id` - The UUID associated with the client device that sent the event _(Only included for events sent from client-side libraries)_
+-   `$distinct_id` - The distinct identifier for whichever user sent the event (Email, UUID, etc.). This can be set by the sender or is defaulted to `$anon_distinct_id` if it is not set
+
+To determine what to do at this stage, we need to make a call to PostgreSQL to determine which scenario we are in:
+
+|                                                                                            |                                                                                                                    |
+| ------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------ |
+| 1. Neither `$anon_distinct_id` nor `$distinct_id` have been associated with a Person       | Create a new Person and add a mapping in PostgreSQL to associate this `$distinct_id` with the new `person_id`      |
+| 2. _Only one_ of `$anon_distinct_id` and `$distinct_id` have been associated with a Person | Create a new mapping to associate the `$distinct_id` and `$anon_distinct_id` with the already existing `person_id` |
+| 3. _Both_ `$anon_distinct_id` _and_ `$distinct_id` have been associated with a Person      | We will [merge these two people](#merging-two-persons) and associate all future events with the first Person       |
+
+> **Note**: In the case of an event sent from the backend, there will be no `$anon_distinct_id` field present. Instead, the same lookups and checks will be performed, but only on the `$distinct_id` which is required for all backend events
```

Comment:
hmm... that's not accurate: what we do here is if there's no anon_distinct_id we just jump to updating properties (well actually we jump to `1.3 - All other events` as we might still need to create the person if they don't exist)
code: https://github.com/PostHog/posthog/blob/4f648268f27989dc10212f96b2b5b2c23e171c53/plugin-server/src/worker/ingestion/person-state.ts#L252 this is not true, so we do noting in the merging path